src/main.rs

```diff
@@ -160,3 +152,30 @@ fn pass_to_sub(command: String) -> Result<(), MainError> {
         None => MainError::Pn(PnError::UnexpectedTermination { command }),
     })
 }
+
+fn get_prepended_path_env(prepend_path: &str) -> OsString {
+    if let Some(path) = env::var_os("PATH") {
+        let mut new_paths: Vec<PathBuf> = vec![PathBuf::from(prepend_path)];
+        let mut current_paths = env::split_paths(&path).collect::<Vec<_>>();
+        new_paths.append(&mut current_paths);
+
+        let new_path = env::join_paths(new_paths.iter()).expect("Failed to prepend path");
+        new_path
```

Comment:
Use iterator to avoid unnecessary allocations:

```rust
        let new_paths = std::iter::once(PathBuf::from(prepend_path));
        let current_paths = env::split_paths(&path);
        let new_paths = new_paths.chain(current_paths);
        env::join_paths(new_paths).expect("Failed to prepend path") // TODO: propagate JoinPathError to main as a meaningful error message
```

The code above can be refactored into:

```rust
        prepend_path
            .pipe(PathBuf::from)
            .pipe(std::iter::once)
            .chain(env::split_paths(&path))
            .pipe(env::join_paths)
            .expect("Failed to prepend path") // TODO: propagate JoinPathError to main as a meaningful error message
```

For your convenience, you may apply the suggested change directly:

```suggestion
        prepend_path
            .pipe(PathBuf::from)
            .pipe(std::iter::once)
            .chain(env::split_paths(&path))
            .pipe(env::join_paths)
            .expect("Failed to prepend path") // TODO: propagate JoinPathError to main as a meaningful error message
```

Reply:
BTW, I added a TODO comment to remind us to remove the `.expect()` later, but if you want to deal with it now, that would be welcomed too.